### PR TITLE
[Cleanup] Balancer Split Files and Document

### PR DESCRIPTION
**Pull request by bh2smith**
event handler file had become quite large with a logical separation into 4

1. event handler: itself (responsible for definition and implementation of `BalancerPoolRegistry`)
2. pool_storage: all structs/models for the module including PoolStorage (this is analogous to the database file for GPv2Settlement event listening). 
3. pool_info_fetching: this include a trait and one implementation of the trait.
4. pool_fetching: unchanged.

No logic changes, but please check that all things which were made public are indeed correct.

Note that I have also documented the module and each file here as well.

### Test Plan
existing CI.

### CODE REVIEW

**Review by vkgnosis — APPROVED** [review]

File: shared/src/balancer/event_handler.rs [hunk 1]
```diff
@@ -355,17 +130,20 @@ impl EventStoring<WeightedPool2TokensFactoryEvent> for PoolStorage {
         events: Vec<EthContractEvent<WeightedPool2TokensFactoryEvent>>,
         range: RangeInclusive<BlockNumber>,
     ) -> Result<()> {
-        let balancer_events = convert_two_token_pool_created(events)?;
-        PoolStorage::replace_events_inner(self, range.start().to_u64(), balancer_events).await?;
-        Ok(())
+        PoolStorage::replace_events_inner(
+            self,
```
Comment (nlordell) [inline shared/src/balancer/event_handler.rs:158]: I missed this before but this can likely be written as `self.replace_events_inner(...)`.

**Review by nlordell — APPROVED** [review]
Approving assuming there were no code/logic changes besides what was required for moving files around.

File: shared/src/balancer/event_handler.rs (hunk 1 above)
Comment (bh2smith) [reply shared/src/balancer/event_handler.rs:158]: Yea probably, will check it out.

**Review by fedgiac — APPROVED** [review]
The file split makes sense and the docs are exhaustive.

Note that `logexpmath` is going to be moved in a subdirectory, so it's ok to skip its documentation.

**Pull request merged**